src/mds/MDCache.cc

```diff
@@ -10865,6 +10945,16 @@ void MDCache::decode_replica_inode(CInode *&in, bufferlist::const_iterator& p, C
     if (!dn->get_linkage()->is_primary() || dn->get_linkage()->get_inode() != in)
       dout(10) << __func__ << " different linkage in dentry " << *dn << dendl;
   }
+
+  if (struct_v >= 2) {
+    __u32 state;
+    decode(state, p);
+    state &= CInode::MASK_STATE_REPLICATED;
```

Comment:
Why can't we just bitwise & with `CInode::STATE_RANDEPHEMERALPIN` . I didn't get why `CInode::MASK_STATE_REPLICATED` is defined separately (but using the value of `CInode::STATE_RANDEPHEMERALPIN`)?

Reply:
> Why can't we just bitwise & with `CInode::STATE_RANDEPHEMERALPIN`

I'm following the same logic as for the exported/import case, e.g. `CInode::decode_import`. We can't just `state_set` though for random pins because it needs added to the random pin set as well. That work is all done in the `set_ephemeral_rand`.

Reply:
Ack. I wasn't questioning the logic though. Just was slightly confused why define and use CInode::MASK_STATE_REPLICATED when we already have CInode::STATE_RANDEPHEMERALPIN. But yea, it doesn't sound good to use CInode::STATE_RANDEPHEMERALPIN as a placeholder for replicated inode.